File: bank.cpp
#include <iostream>
#include "bank.h"

using namespace std;

Bank::Bank() : name{"Volksbank"} {}

Bank::Bank(string &name) : name{name} {};

Bank::Bank(string &name, vector<BankUser> &bankUserRecord) : bankUserRecord{bankUserRecord}, name{name} {};

string Bank::getName()
{
    return name;
}

vector<BankUser> Bank::getBankUserRecord()
{
    return bankUserRecord;
}

BankUser &Bank::getBankUser(int index)
{
    return bankUserRecord[index];
}

void Bank::addBankUser(string &userName)
{
    bankUserRecord.push_back(userName);
}

File: Bank.java
import java.util.Vector;
public class Bank {
    Vector<BankUser> bankUserRecord;
    String name;

    public Bank(){name="Volkswagen";}

    public Bank(String name) {
        this.name = name;
    }

    public Bank(Vector<BankUser> bankUserRecord, String name) {
        this.bankUserRecord = bankUserRecord;
        this.name = name;
    }

    public Vector<BankUser> getBankUserRecord() {
        return bankUserRecord;
    }

    public String getName() {
        return name;
    }

    public BankUser getBankUserRecord(int index) {
        return bankUserRecord.elementAt(index);
    }

    public void addBankUser(BankUser userName) {
        bankUserRecord.add(userName);
    }
}
